peer.go

```diff
@@ -402,6 +408,12 @@ func (p *peer) handleVersionMsg(msg *wire.MsgVersion) {
 	// Set the remote peer's user agent.
 	p.userAgent = msg.UserAgent
 
+	// Set the peer's time offset.
+	p.timeOffset = msg.Timestamp.Unix() - time.Now().Unix()
+
+	// Set the peer's ID.
+	p.id = atomic.AddInt32(&nodeCount, 1)
```

Comment:
If you're going to use atomic, the variable must be to aligned in the struct.  Thus, it needs to be moved up after `disconnect` before the `net.Conn` interface.  Also, it needs a comment after it clearly stating it's used by atomic so people know it can't be moved.